lacommunaute/forum_conversation/tests/tests_models.py

```diff
@@ -41,34 +38,32 @@ class TopicManagerTest(TestCase):
     def test_unanswered(self):
         forum = ForumFactory()
 
-        TopicFactory(forum=forum, posts_count=0)
-        topic = TopicFactory(forum=forum, posts_count=1)
-        TopicFactory(forum=forum, posts_count=2)
+        TopicFactory(forum=forum)
+        topic = TopicFactory(forum=forum, with_post=True)
+        TopicFactory(forum=forum, with_post=True, answered=True)
 
-        TopicFactory(forum=forum, posts_count=1, status=Topic.TOPIC_LOCKED)
-        TopicFactory(forum=forum, posts_count=1, type=Topic.TOPIC_ANNOUNCE)
-        TopicFactory(forum=forum, posts_count=1, approved=False)
+        TopicFactory(forum=forum, with_post=True, status=Topic.TOPIC_LOCKED)
+        TopicFactory(forum=forum, with_post=True, type=Topic.TOPIC_ANNOUNCE)
+        TopicFactory(forum=forum, with_disapproved_post=True)
 
         self.assertEqual(Topic.objects.unanswered().get(), topic)
 
     def test_unanswered_order(self):
         forum = ForumFactory()
-        last_post_dates = [datetime(2025, 5, i, tzinfo=timezone.utc) for i in range(20, 24)]
-
-        TopicFactory.create_batch(
-            size=len(last_post_dates), forum=forum, posts_count=1, last_post_on=Iterator(last_post_dates)
-        )
-        assert list(Topic.objects.unanswered().values_list("last_post_on", flat=True)) == last_post_dates[::-1]
+        topics = TopicFactory.create_batch(size=3, forum=forum, with_post=True)
+        expected_date_list = [topic.last_post_on for topic in topics[::-1]]
```

Comment:
Un poil plus clair :shrug: 
```suggestion
        expected_date_list = [topic.last_post_on for topic in reversed(topics)]
```